config/initializers/commontator.rb

```diff
@@ -1,187 +1,249 @@
+  # Note: Gravatar has several security implications for your users
+  #       It makes your users trackable across different sites and
+  #       allows de-anonymization attacks against their email addresses
+  #       If you absolutely want to keep users' email addresses or identities secret,
+  #       do not use Gravatar or similar services
+  # Default: lambda { |user, view|
+  #            view.commontator_gravatar_image_tag(
+  #              user, 1, :s => 60, :d => 'mm') }
+  config.user_avatar_proc = lambda { |user, view|
+                                     view.commontator_gravatar_image_tag(
+                                       user, 1, :s => 60, :d => 'mm') }
+
+  # user_email_proc
+  # Type: Proc
+  # Arguments: a user (acts_as_commontator), a mailer (ActionMailer::Base)
+  # Returns: the user's email address (String)
+  # The default works for Devise's defaults
+  # If the mailer argument is nil, Commontator intends to hash the email and send the hash
+  # to Gravatar, so you should always return the user's email address (if using Gravatar)
+  # If the mailer argument is not nil, then Commontator intends to send an email to
+  # the address returned; you can prevent it from being sent by returning a blank String
+  # Default: lambda { |user, mailer| user.try(:email) || '' }
+  config.user_email_proc = lambda { |user, mailer| user.try(:email) || '' }
+
+
+
+  # Thread/Commontable (acts_as_commontable) Configuration
+
+  # comment_filter
+  # Type: Arel node (Arel::Nodes::Node) or nil
+  # Arel that filters visible comments
+  # If specified, visible comments will be filtered according to this Arel node
+  # A value of nil will cause no filtering to be done
+  # Moderators can manually override this filter for themselves
+  # Example: Commontator::Comment.arel_table[:deleted_at].eq(nil) (hides deleted comments)
+  #          This is not recommended, as it can cause confusion over deleted comments
+  #          If using pagination, it can also cause comments to change pages
+  # Default: nil (no filtering - all comments are visible)
+  config.comment_filter = nil
+
+  # thread_read_proc
+  # Type: Proc
+  # Arguments: a thread (Commontator::Thread), a user (acts_as_commontator)
+  # Returns: a Boolean, true iif the user should be allowed to read that thread
+  # Note: can be called with a user object that is nil (if they are not logged in)
+  # Default: lambda { |thread, user| true } (anyone can read any thread)
+  config.thread_read_proc = lambda { |thread, user| true }
+
+  # thread_moderator_proc
+  # Type: Proc
+  # Arguments: a thread (Commontator::Thread), a user (acts_as_commontator)
+  # Returns: a Boolean, true iif the user is a moderator for that thread
+  # If you want global moderators, make this proc true for them regardless of thread
+  # Default: lambda { |thread, user| false } (no moderators)
+  config.thread_moderator_proc = lambda { |thread, user| false }
```

Comment:
Unused block argument - `thread`. If it's necessary, use `_` or `_thread` as an argument name to indicate that it won't be used. Also consider using a proc without arguments instead of a lambda if you want it to accept any arguments but don't care about them.<br>Unused block argument - `user`. If it's necessary, use `_` or `_user` as an argument name to indicate that it won't be used. Also consider using a proc without arguments instead of a lambda if you want it to accept any arguments but don't care about them.